controllers/nodemaintenance_controller.go

```diff
@@ -55,16 +72,379 @@ type NodeMaintenanceReconciler struct {
+				if err := invalidateLease(r.Client, nodeName); err != nil {
+					return err
+				}
+			}
+			return nil
+		}
+		return err
+	}
+	return r.stopNodeMaintenanceImp(node)
+}
+
+func (r *NodeMaintenanceReconciler) fetchNode(nodeName string) (*corev1.Node, error) {
+	node, err := r.drainer.Client.CoreV1().Nodes().Get(context.Background(), nodeName, metav1.GetOptions{})
+	if err != nil && errors.IsNotFound(err) {
+		r.logger.Error(err, "Node cannot be found", "nodeName", nodeName)
+		return nil, err
+	} else if err != nil {
+		r.logger.Error(err, "Failed to get node", "nodeName", nodeName)
+		return nil, err
+	}
+	return node, nil
+}
+
+func (r *NodeMaintenanceReconciler) initMaintenanceStatus(nm *nodemaintenancev1beta1.NodeMaintenance) error {
+	if nm.Status.Phase == "" {
+		nm.Status.Phase = nodemaintenancev1beta1.MaintenanceRunning
+		pendingList, errlist := r.drainer.GetPodsForDeletion(nm.Spec.NodeName)
+		if errlist != nil {
+			return fmt.Errorf("Failed to get pods for eviction while initializing status")
+		}
+		if pendingList != nil {
+			nm.Status.PendingPods = GetPodNameList(pendingList.Pods())
+		}
+		nm.Status.EvictionPods = len(nm.Status.PendingPods)
+
+		podlist, err := r.drainer.Client.CoreV1().Pods(metav1.NamespaceAll).List(
+			context.Background(),
+			metav1.ListOptions{
+				FieldSelector: fields.SelectorFromSet(fields.Set{"spec.nodeName": nm.Spec.NodeName}).String(),
+			})
+		if err != nil {
+			return err
+		}
+		nm.Status.TotalPods = len(podlist.Items)
+		err = r.Client.Status().Update(context.TODO(), nm)
+		return err
+	}
+	return nil
+}
+
+func (r *NodeMaintenanceReconciler) onReconcileErrorWithRequeue(nm *nodemaintenancev1beta1.NodeMaintenance, err error, duration *time.Duration) (reconcile.Result, error) {
+	nm.Status.LastError = err.Error()
+
+	if nm.Spec.NodeName != "" {
+		pendingList, _ := r.drainer.GetPodsForDeletion(nm.Spec.NodeName)
+		if pendingList != nil {
+			nm.Status.PendingPods = GetPodNameList(pendingList.Pods())
+		}
+	}
+
+	updateErr := r.Client.Status().Update(context.TODO(), nm)
+	if updateErr != nil {
+		r.logger.Error(updateErr, "Failed to update NodeMaintenance with \"Failed\" status")
+	}
+	if duration != nil {
+		r.logger.Info(FixedDurationReconcileLog)
+		return reconcile.Result{RequeueAfter: *duration}, nil
+	}
+	r.logger.Info("Reconciling with exponential duration")
+	return reconcile.Result{}, err
+}
+
+func (r *NodeMaintenanceReconciler) onReconcileError(nm *nodemaintenancev1beta1.NodeMaintenance, err error) (reconcile.Result, error) {
+	return r.onReconcileErrorWithRequeue(nm, err, nil)
+
+}
+
+//// set readable timestamp format in logs, not reference time
+//func init() {
+//	log.SetFormatter(&log.TextFormatter{TimestampFormat: "2006-01-02 15:04:05.000000", FullTimestamp: true})
+//}
+
+// writer implements io.Writer interface as a pass-through for klog.
+type writer struct {
+	logFunc func(args ...interface{})
+}
+
+// Write passes string(p) into writer's logFunc and always returns len(p)
```

Comment:
is this being used? doesn't look like related to the controller package.

Reply:
yes, it is used: it implements the writer interface for the struct above by passing the bytes to a logger. The struct is initialized and used for the out and error streams of the "drainer", see bottom of `initDrainer` func
Does it belong here? Probably not. As other code also doesn't. But I didn't want to do more refacorings than needed for the sdk update in this PR 🤷🏼‍♂️